runtime/ms-rest/lib/webResource.js

```diff
@@ -163,7 +163,7 @@ class WebResource {
       }
       let baseUrl = options.baseUrl;
       let url = baseUrl + (baseUrl.endsWith('/') ? '' : '/') + (options.pathTemplate.startsWith('/') ? options.pathTemplate.slice(1) : options.pathTemplate);
-      let segments = url.match(/({\w*\s*\w*})/ig);
+      let segments = url.match(/({\W*\S*\W*})/ig);
```

Comment:
Yeah, not sure I understand it either. Is "-" the only special char we want to allow or do we allow others? if so, should we just have the expression accept "-" between words? 
I'm not sure about the original regex either though, do we allow spaces in between words for the property inside {}?